control/controlclient/noise.go

```diff
@@ -116,14 +124,17 @@ func NewNoiseClient(opts NoiseOpts) (*NoiseClient, error) {
 	}
 	var httpPort string
 	var httpsPort string
-	if u.Port() != "" {
+	if port := u.Port(); port != "" {
 		// If there is an explicit port specified, trust the scheme and hope for the best
 		if u.Scheme == "http" {
-			httpPort = u.Port()
+			httpPort = port
 			httpsPort = "443"
+			if (testenv.InTest() || controlIsPlaintext()) && (u.Hostname() == "127.0.0.1" || u.Hostname() == "localhost") {
```

Comment:
Do we need the `testenv.InTest` if we have the envknob?

Reply:
Yes. InTest are for normal tests (in corp). The envknob is only for tstest/integration tests where tailscaled runs as a child process that doesn't look like a test to itself.